### PR TITLE
Put prometheus and loki auth under the basicAuth object

### LINKED ISSUE #52: Put username and password underneath basicAuth section
For prometheus and loki services, we have the structure:
```
host:
username:
password:
```

That just kinda assumes basic authentication, but makes it unclear how to support alternative authentication methods.

### PR DESCRIPTION
**Pull request by petewall**
Also makes usernames and passwords optional.

Fixes #52

### CODE REVIEW

File: charts/k8s-monitoring/templates/agent_config/_loki.river.txt [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
   endpoint {
-    url = nonsensitive(local.file.loki_host.content) + "{{ .Values.externalServices.loki.writeEndpoint }}"
-
+    url = nonsensitive(local.file.loki_host.content) + "{{ .writeEndpoint }}"
+{{ if .basicAuth }}
     basic_auth {
-      username = local.file.loki_username.content
-      password = local.file.loki_password.content
+      {{ if .basicAuth.username }}username = local.file.loki_username.content{{ end }}
+      {{ if .basicAuth.password }}password = local.file.loki_password.content{{ end }}
```
Comment (skl) [inline charts/k8s-monitoring/templates/agent_config/_loki.river.txt:25]: Is it necessary to guard the username and password with individual conditionals or is it enough just to have the `if .basicAuth` check?

File: charts/k8s-monitoring/templates/credentials.yaml [hunk 2]
```diff
@@ -7,11 +7,19 @@ type: Opaque
 data:
 {{- if .Values.metrics.enabled }}
   prometheus_host: {{ required ".Values.externalServices.prometheus.host is required to use metrics. Please set it and try again." .Values.externalServices.prometheus.host | b64enc | quote }}
-  prometheus_username: {{ required ".Values.externalServices.prometheus.username is required to use metrics. Please set it and try again."  .Values.externalServices.prometheus.username | toString | b64enc | quote }}
-  prometheus_password: {{ required ".Values.externalServices.prometheus.password is required to use metrics. Please set it and try again."  .Values.externalServices.prometheus.password | toString | b64enc | quote }}
+  {{- if .Values.externalServices.prometheus.basicAuth.username }}
+  prometheus_username: {{ .Values.externalServices.prometheus.basicAuth.username | toString | b64enc | quote }}
+  {{- end }}
+  {{- if .Values.externalServices.prometheus.basicAuth.username }}
```
Comment (skl) [inline charts/k8s-monitoring/templates/credentials.yaml:13]:
```suggestion
  {{- if .Values.externalServices.prometheus.basicAuth.password }}
```

File: charts/k8s-monitoring/templates/credentials.yaml [hunk 3]
```diff
@@ -7,11 +7,19 @@ type: Opaque
...
+  {{- end }}
 {{- end }}
 {{- if .Values.logs.enabled }}
   loki_host: {{ required ".Values.externalServices.loki.host is required to use logs or events. Please set it and try again."  .Values.externalServices.loki.host | b64enc | quote }}
-  loki_username: {{ required ".Values.externalServices.loki.username is required to use logs or events. Please set it and try again."  .Values.externalServices.loki.username | toString | b64enc | quote }}
-  loki_password: {{ required ".Values.externalServices.loki.password is required to use logs or events. Please set it and try again."  .Values.externalServices.loki.password | toString | b64enc | quote }}
+  {{- if .Values.externalServices.loki.basicAuth.username }}
+  loki_username: {{ .Values.externalServices.loki.basicAuth.username | toString | b64enc | quote }}
+  {{- end }}
+  {{- if .Values.externalServices.loki.basicAuth.username }}
```
Comment (skl) [inline charts/k8s-monitoring/templates/credentials.yaml:22]:
```suggestion
  {{- if .Values.externalServices.loki.basicAuth.password }}
```

File: charts/k8s-monitoring/templates/agent_config/_loki.river.txt (hunk 1 above)
Comment (petewall) [reply charts/k8s-monitoring/templates/agent_config/_loki.river.txt:25]: If password is set, but username is not, I want only one to be created.

**Review by skl — APPROVED** [review]

**Pull request merged**